agents/agent_loop.py

```
@@ -1,28 +1,149 @@
+                
+                if self.llm:
+                    # Using run_in_executor for blocking LLM call
+                    loop = asyncio.get_running_loop()
+                    try:
+                        decision = await loop.run_in_executor(None, self.llm.predict, prompt)
+                        if "yes" in decision.lower():
+                            self.record_to_contract(mined, rate)
+                    except Exception as e:
+                        print(f"[ELIZA] LLM error: {e}")
+                else:
+                     # Fallback if LLM not available
```

Flake8 (7.3.0)

[error] indentation is not a multiple of 4 (comment)

(E114)